Fix evohome HVAC modes for VisionPro Wifi systems

zxdavb opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This PR fixes a bug where some systems cannot set the HVAC mode and includes tests that confirm the bug is fixed.

Other tests are affected only in that this PR changes the function that calls  `async_setup_component()` into a generator.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

VisionPro Wifi systems are compatible with Honeywell (now Resideo) TCC, but have slightly different operating modes (e.g. `Off` instead of `HeatingOff`). This was mentioned in #32585 (which was closed when the primary issue was addressed in #32604).

Thus, these specific systems cannot be turned on/off via the UI or with the usual service calls. 

NOTE: This fix adds a test for service calls (that exposed the bug, and confirms it has been resolved). 

This new test includes some changes to  **conftest.py**, and so some changes to other tests. 


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

zxdavb committed:
Squashed commit of the following:

commit a6651d888c7b5c1e095167bc4835b9e8baf7c39a
Merge: 359a53a160 a923f15d17
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sun Sep 22 00:06:35 2024 +0100

    commit

commit 359a53a16052006de0046e139f73e2ca59026025
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 23:29:41 2024 +0100

    Rename some evohome constants for clarity / readability (#126394)

    initial commit

commit 553e9bc2f0a24954cd819f9091edd33a5f0efa0f
Author: Erik Montnemery <erik@montnemery.com>
Date:   Sat Sep 21 23:14:12 2024 +0200

    Add issue asking users to disable ESPHome assist_in_progress sensor (#125805)

    * Add issue asking users to disable ESPHome assist_in_progress binary sensor

    * Include integration name in title and description

    * Add repair flow

    * Improve test coverage

commit 6792aeaa3838d9b762f3206628fe33e3dbc587de
Author: Michael <35783820+mib1185@users.noreply.github.com>
Date:   Sat Sep 21 21:11:17 2024 +0200

    Set connection and command timeout in VLC Telnet (#126401)

    use 1s lower than scan interval

commit 45d50212592cc9cdd71790fc3f950ea7ebc19b1c
Author: Allen Porter <allen@thebends.org>
Date:   Sat Sep 21 10:56:13 2024 -0700

    Update the Google Photos integration to limit scope to Home Assistant created content (#126398)

commit f2221093de2248010df62d23bb92c6647a158cf6
Author: Steven B. <51370195+sdb9696@users.noreply.github.com>
Date:   Sat Sep 21 18:03:51 2024 +0100

    Fix tplink number platform to use intended BOX mode (#126397)

    The NumberMode should be BOX as per the entity description but due to the missing dataclass decorator was resolving to NumberMode.AUTO.

commit eddb6986830a1c9808884bdc8426133ed1526104
Merge: 5a9a19cf6c 24106114b4
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 15:47:58 2024 +0100

    Merge remote-tracking branch 'upstream/dev' into evo_fix_visionpro

commit 5a9a19cf6cff827713f41091205ce7bf8ff45366
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 13:37:29 2024 +0100

    un-rename symbols

commit dfe44a168485c859bf3837da73273d74c45f79ee
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 11:40:19 2024 +0100

    move attr to all children of TCS

commit f16565e892e608ee6c4ff3ceba712cc446174586
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 11:22:12 2024 +0100

    remove tests (for now)

commit 8936c7b6962cc91517c553d962d6e31ece9a9ac4
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 11:19:40 2024 +0100

    tweaks to codes

commit be711f56d915bb14691edafaf97a785d503b1d15
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 09:31:29 2024 +0100

    Add tests of services and entity methods

commit 862b097345728fa247b95e796030c596c42ff7b6
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 09:21:52 2024 +0100

    Refactor evohome for visionpro modes,

commit 5eec2d0a248cb5e953591133fc59e8bd9687b87c
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 09:20:19 2024 +0100

    Improve test suit - setup/teardown, use generator, rename symbols

zxdavb committed:
Squashed commit of the following:

commit a6651d888c7b5c1e095167bc4835b9e8baf7c39a
Merge: 359a53a160 a923f15d17
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sun Sep 22 00:06:35 2024 +0100

    commit

commit 359a53a16052006de0046e139f73e2ca59026025
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 23:29:41 2024 +0100

    Rename some evohome constants for clarity / readability (#126394)

    initial commit

commit 553e9bc2f0a24954cd819f9091edd33a5f0efa0f
Author: Erik Montnemery <erik@montnemery.com>
Date:   Sat Sep 21 23:14:12 2024 +0200

    Add issue asking users to disable ESPHome assist_in_progress sensor (#125805)

    * Add issue asking users to disable ESPHome assist_in_progress binary sensor

    * Include integration name in title and description

    * Add repair flow

    * Improve test coverage

commit 6792aeaa3838d9b762f3206628fe33e3dbc587de
Author: Michael <35783820+mib1185@users.noreply.github.com>
Date:   Sat Sep 21 21:11:17 2024 +0200

    Set connection and command timeout in VLC Telnet (#126401)

    use 1s lower than scan interval

commit 45d50212592cc9cdd71790fc3f950ea7ebc19b1c
Author: Allen Porter <allen@thebends.org>
Date:   Sat Sep 21 10:56:13 2024 -0700

    Update the Google Photos integration to limit scope to Home Assistant created content (#126398)

commit f2221093de2248010df62d23bb92c6647a158cf6
Author: Steven B. <51370195+sdb9696@users.noreply.github.com>
Date:   Sat Sep 21 18:03:51 2024 +0100

    Fix tplink number platform to use intended BOX mode (#126397)

    The NumberMode should be BOX as per the entity description but due to the missing dataclass decorator was resolving to NumberMode.AUTO.

commit eddb6986830a1c9808884bdc8426133ed1526104
Merge: 5a9a19cf6c 24106114b4
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 15:47:58 2024 +0100

    Merge remote-tracking branch 'upstream/dev' into evo_fix_visionpro

commit 5a9a19cf6cff827713f41091205ce7bf8ff45366
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 13:37:29 2024 +0100

    un-rename symbols

commit dfe44a168485c859bf3837da73273d74c45f79ee
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 11:40:19 2024 +0100

    move attr to all children of TCS

commit f16565e892e608ee6c4ff3ceba712cc446174586
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 11:22:12 2024 +0100

    remove tests (for now)

commit 8936c7b6962cc91517c553d962d6e31ece9a9ac4
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 11:19:40 2024 +0100

    tweaks to codes

commit be711f56d915bb14691edafaf97a785d503b1d15
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 09:31:29 2024 +0100

    Add tests of services and entity methods

commit 862b097345728fa247b95e796030c596c42ff7b6
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 09:21:52 2024 +0100

    Refactor evohome for visionpro modes,

commit 5eec2d0a248cb5e953591133fc59e8bd9687b87c
Author: David Bonnes <zxdavb@bonnes.me>
Date:   Sat Sep 21 09:20:19 2024 +0100

    Improve test suit - setup/teardown, use generator, rename symbols

zxdavb committed:
Merge branch 'dev' into evo_fix_vispro

somdoron committed:
Add fan support for KNX climate entities (#126368)

* Add fan mode support to knx climate

* fix linting errors

* remove unneeded None protection from CONF_FAN_PERCENTAGES_MODES

* Update homeassistant/components/knx/climate.py

Co-authored-by: Matthias Alphart <farmio@alphart.net>

* Update homeassistant/components/knx/climate.py

Co-authored-by: Matthias Alphart <farmio@alphart.net>

* Update homeassistant/components/knx/climate.py

Co-authored-by: Matthias Alphart <farmio@alphart.net>

* Update homeassistant/components/knx/schema.py

Co-authored-by: Matthias Alphart <farmio@alphart.net>

* find closest percentage when not in fan modes

* new field for fan speed mode, max steps apply to both step and percentage

* not picking FAN_OFF when the percentage is closest to zero

* add fan zero mode to support auto mode

* use StrEnum for FanZeroMode

* change default to 'percent'

* fix mypy errors

---------

Co-authored-by: Matthias Alphart <farmio@alphart.net>

starkillerOG committed:
Use insecure SSL cipher for Reolink aiohttp clientsession (#126687)

jvmahon committed:
Add button platform to Matter integration (#123665)

* Add files via upload

* add test

* add discovery schemas for operational state commands

* tests

* add filter resets

* add filter reset buttons

* Apply suggestions from code review

* tweak test

---------

Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

civita committed:
Update strings in tailscale (#124143)

zxdavb committed:
Merge remote-tracking branch 'upstream/dev' into evo_fix_vispro

joostlek committed:
Add coordinator to Spotify (#123548)

joostlek committed:
Improve typing in Yamaha (#123982)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

gjohansson-ST committed:
Add diagnostics to Workday (#126691)

bdraco committed:
Bump aiohttp to 3.10.6 (#126690)

allenporter committed:
Improve Roborock error handling (#124267)

starkillerOG committed:
Remove leftover wrong icon from Reolink (#126698)

Remove wrong icon

Commit:
Add sensor platform to Bring integration (#126642)

* Add sensor platform to Bring integration

* Add more tests

* unignore typedef check

* Update language sensor

* update snapshot

* changes

* add entities

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* add units

* lowercase

* snapshot

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

emontnemery committed:
Add tests which directly test the recorder job wrappers (#125338)

Commit:
Add diagnostics platform to Bring integration (#126695)

epenet committed:
Avoid raw string in device_tracker source_type (#126601)

joostlek committed:
Add Spelling Bee and connections support to NYT Games (#126567)

joostlek committed:
Add Aqara brand (#126658)

bdraco committed:
Bump google-generativeai to 0.8.2 (#126696)

changelog: https://github.com/google-gemini/generative-ai-python/compare/v0.7.2...v0.8.2

Commit:
Update snapshot for Bring tests (#126699)

epenet committed:
Add specific EntityDescription to describe device tracker entities (#126586)

* Add TrackerEntityDescription to describe tracker entities

* Improve

* Adjust components

* Add ScannerEntityDescription

* Simplify

* Revert

* Set TrackerEntity default source type to SourceType.GPS

* Fix rebase

* Adjust default

* Remove source_type from EntityDescription

* Fix rebase

* Docstring

* Remove BaseTrackerEntityDescription

allenporter committed:
Bump python-google-photos-library-api to 0.12.1 (#126709)

tal committed:
Update Tuya integration with target distance (#126700)

Co-authored-by: Franck Nijhof <git@frenck.dev>

epenet committed:
Remove redundant source_type property from TrackerEntities (#126717)

epenet committed:
Remove redundant source_type property from ScannerEntities (#126716)

ol-iver committed:
Update `denonavr` to `v1.0.0` (#126703)

dependabot[bot] committed:
Bump github/codeql-action from 3.26.8 to 3.26.9 (#126715)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bdraco committed:
Bump zeroconf to 0.135.0 (#126706)

Bre77 committed:
Add Storm watch active to Tesla integrations (#126704)

cdce8p committed:
Fix pytest-asyncio DeprecationWarning (#126718)

elmurato committed:
Add workaround to avoid blocking imports by dnspython (#121702)

epenet committed:
Move pylint decorator plugin and add tests (#126719)

joostlek committed:
Introduce snapshot testing to matter (#126693)

* Introduce snapshot testing to matter

* Introduce snapshot testing to matter

edenhaus commented on tests/components/evohome/test_init.py:
```diff
@@ -23,8 +23,9 @@ async def test_entities(
 
     # some extended state attrs are relative the current time
-    freezer.move_to("2024-07-10 12:00:00+00:00")
+    freezer.move_to("2024-07-10T12:00:00Z")
```
Why are you changing this time?

zxdavb replied:
I used a different format, only to be consistent with the dtm string used across the evohome landscape...

Is there a preferred format to favour?

zxdavb replied:
Oh, sorry - I think I misunderstood your question.

The entities have extended state attrs that include scheduled setpoints (i.e. a dtm : target temp pair). Unless the day of week / time of day if fixed here, then they will not likely match those in the snapshots.

edenhaus commented on tests/components/evohome/test_init.py:
```diff
@@ -23,8 +23,9 @@ async def test_entities(
 
-    await setup_evohome(hass, config, install=install)
+    async for _ in setup_evohome(hass, config, install=install):
+        await hass.async_block_till_done()
```
Why do we need to wait n times instead of only one at the end?

zxdavb replied:
This is a long story...

In `async_setup()` there is one or two `hass.async_create_task(async_load_platform(...` and the `async_block_till_done()` in `setup_evohome()` definitely blocks until all the `async_setup_platform()` calls are complete.

However, when `async_setup_platform()` invokes the `async_add_entities` callback, it includes `update_before_add=True`, which causes `ep._async_add_and_update_entities()` to be invoked, instead of `ep._async_add_entities()`.

The former creates further tasks, one for each entity, and then:
```
async with self.hass.timeout.async_timeout(timeout, self.domain):
    results = await asyncio.gather(*tasks, return_exceptions=True)
```

Unfortunately, the first `async_block_till_done()` appears to return *before* this second tranche of tasks are DONE.

zxdavb replied:
However, in `test_entities` of **test_init.py**, you notice another.

In `async_setup()` there is one or two `hass.async_create_task(async_load_platform(...` and the `async_block_till_done()` in `setup_evohome()` definitely blocks until all the corresponding `async_setup_platform()` calls are complete.

However, when `async_setup_platform()` invokes the `async_add_entities` callback, it includes `update_before_add=True`, which causes `ep._async_add_and_update_entities()` to be invoked, instead of `ep._async_add_entities()`.

The former creates further tasks, one for each entity, and then:
```
async with self.hass.timeout.async_timeout(timeout, self.domain):
    results = await asyncio.gather(*tasks, return_exceptions=True)
```

Unfortunately, the first `async_block_till_done()` appears to return *before* this second tranche of tasks are DONE.

edenhaus commented on tests/components/evohome/conftest.py:
```diff
@@ -117,18 +119,29 @@ def config() -> dict[str, str]:
+        locn = user_locations_config_fixture(install)[0]
+
+    utc_offset: int = locn["locationInfo"]["timeZone"]["currentOffsetMinutes"]  # type: ignore[assignment, call-overload, index]
+    dt_util.set_default_time_zone(timezone(timedelta(minutes=utc_offset)))
```
Why do we need to set the timezone here?

zxdavb replied:
The date time strings in the vendor's JSON are (local) time zone naive. 

By switching the default time zone to the same as that used by each evohome fixture, it's much easier to compare the calculated date time strings against those in the fixtures.

zxdavb replied:
See: #126679

edenhaus reviewed (commented):
Can we split this PR into multiple PRs as it includes tests changes, which are not part of the bugfix (like adding tests for all services)

allenporter committed:
Add an entity description for Google Calendar (#125469)

zxdavb commented:
> Can we split this PR into multiple PRs as it includes tests changes, which are not part of the bugfix (like adding tests for all services)

I'll split this out to multiple PRs...

emontnemery committed:
Make statistics validation create issue registry issues (#122595)

* Make statistics validation create issue registry issues

* Disable creating issue about outdated MariaDB version in tests

* Use call_soon_threadsafe instead of run_callback_threadsafe

* Update tests

* Fix flapping test

* Disable creating issue about outdated SQLite version in tests

* Implement agreed changes

* Add translation strings for issue titles

* Update test

joostlek committed:
Enable Zwave notification sensors by default (#125326)

* Enable Zwave notification sensors by default

* Enable Zwave notification sensors by default

* Enable Zwave notification sensors by default

* Enable Zwave notification sensors by default

* Enable Zwave notification sensors by default

* Enable Zwave notification sensors by default

* Enable Zwave notification sensors by default

* Enable Zwave notification sensors by default

* Fix the check to (dis)allow discovering a value multiple times

* Prevent discovery of duplicate Notification CC sensors

* alarm sensors disabled by default

* one more fix

* Update diagnostics tests

---------

Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

epenet committed:
Add specific EntityDescription to describe calendar entities (#126726)

piitaya committed:
Use entity selector in Homekit bridge config flow (#126340)

Use entity selector in homekit bridge config flow

marcelveldt committed:
Don't crash entire Matter integration setup when one node is failing (#126491)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

cdce8p committed:
Update isal to 1.7.1 (#126742)

emontnemery committed:
Use 'select' instead of 'click' or 'press' when guiding users in flows (#126731)

cdce8p committed:
Update lxml to 5.3.0 (#126725)

joostlek committed:
Change trigger platform key to trigger (#124357)

* fix

* Fix

* Fix

* Update homeassistant/helpers/config_validation.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Fix

* Fix

* Fix

* Fix

* Add more tests

* Fix

* Fix tests

* Add tests

* Let's see what the CI does

* It fails on the code that tested the thing ofc

* It fails on the code that tested the thing ofc

* Revert test thingy

* Now the test works again, lovely

* Another one

* Fix websocket thingy

* Only copy when needed

* Improve comment

* Remove test

* Fix docstring

* I think this now also work since this transforms trigger to platform

* Add comment

* Update homeassistant/helpers/config_validation.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Update homeassistant/helpers/config_validation.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Update homeassistant/helpers/config_validation.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Check for mapping

* Add test

* Update homeassistant/helpers/config_validation.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Update test to also test for trigger keys

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

epenet committed:
Use shorthand attributes in tesla_fleet device tracker (#126736)

zxdavb committed:
remove TZ code (moved to PR #126679)

joostlek committed:
Make Matter snapshot logic a shared function (#126744)

epenet committed:
Use shorthand attributes in mysensors device tracker (#126738)

epenet committed:
Add pylint checks for fixture scope (#126723)

* Prevent session scope fixtures in component tests

* Link message to the decorator - not the function

* Add checks for package also

* Add check for session scope autouse

* Rename variable

* Adjust message

* Ignore fancy autouse

* Simplify

joostlek committed:
Use snake case in matter fixture nodes (#126743)

joostlek committed:
Update trigger validation message (#126749)

c0ffeeca7 committed:
Use bold to markup UI strings (#126748)

* Use bold to markup UI strings

* Use bold to markup UI strings

epenet committed:
Rename DOMAIN_DATA HassKey constants to DATA_COMPONENT (#126746)

* Rename DOMAIN_DATA HassKey constant to DATA

* DATA -> DATA_COMPONENT

mj23000 committed:
Add sound modes to Bang & Olufsen devices (#121209)

* Add sound mode functionality

* Fix naming

* Change unique sound mode symbol

* Add testing for sound modes

* Add test typing

* Use constants for service call parameters

* Add state assertions

* Remove invalid decorator

* Add valid sound mode check

* Add test for invalid sound mode

bdraco committed:
Fix license check for new aiocache (#126753)

joostlek committed:
Improve trigger schema validation to ask for `trigger` instead of `platform` (#126750)

* Add check for missing trigger

* Fix

* Fix

* Escape

zxdavb committed:
undo 'remove patch'

zxdavb committed:
test only CTL services

zxdavb committed:
remove botched fixture - part 1

zxdavb committed:
removed botched fixture - part 2

zxdavb committed:
undo doctweaks

zxdavb committed:
remove unused helper functions

zxdavb committed:
remove lint

zxdavb committed:
move constant around

zxdavb committed:
remove unused block_till_done

bramkragten committed:
Update frontend to 20240925.0 (#126763)

euandekock committed:
Update const.py to add new Australian Server URL (#126714)

Growatt is now redirecting Australian users to a new server. This adds support for this server.

mj23000 committed:
Remove unnecessary dict .keys() calls from Bang & Olufsen (#126762)

Remove useless .keys() calls

epenet committed:
Use shorthand attributes in locative device tracker (#126740)

puddly committed:
Bump ZHA to 0.0.34 (#126766)

zxdavb committed:
don't relay upon library for fixture data

lboue committed:
Add Valve platform to Matter integration (#123311)

* Create water_valve.py

* Update water_valve.py

ValveEntity

* Update water_valve.py

ValveDeviceClass

* Update water_valve.py

* Update water_valve.py

OperationalStatus

* Update water_valve.py

* Update water_valve.py

Commands

* Update water_valve.py

Platform.VALVE

* Update water_valve.py

* Update water_valve.py

operational_status

* Update water_valve.py

current_valve_position

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

attributes

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Open command

* Match Valve entity methods

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* ruff-format

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

Attributes.CurrentLevel

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

async_set_valve_position

* Update water_valve.py

* Update water_valve.py

Bitmaps

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update homeassistant/components/matter/water_valve.py

Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>

* Update homeassistant/components/matter/water_valve.py

Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update discovery.py to add WaterValve

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update discovery.py

* Update discovery.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Update water_valve.py

* Rename water_valve.py to valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* Create test_valve.py

* Update test_valve.py

* Update test_valve.py

* Update test_valve.py

* Update test_valve.py

* Update test_valve.py

* Update test_valve.py

* Update test_valve.py

* Update test_valve.py

* Update test_valve.py

* Create valve.json

* Update air-purifier.json

* Revert "Update air-purifier.json"

This reverts commit b68dce0ccc81bc6fb1db36191de1c296ce54cac3.

* Update valve.json

* Update valve.json

* Update valve.json

* Update test_valve.py

* Update valve.json

* Update test_valve.py

* Update valve.json

* Update valve.json

* Update valve.json

* Update test_valve.py

* Update valve.py

* Update valve.py

* Update valve.py

* add tests

* cleanup

* Clean up variable

* Format

* add tests for state updates

* adjust

* add tests for position

---------

Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Thomas55555 committed:
Bump aioautomower to 2024.9.3 (#126769)

* Bump aioautomower to 2024.9.3

* tests

emontnemery committed:
Remove unignore flow (#126765)

frenck committed:
Bump version to 2024.11.0dev0 (#126776)

gjohansson-ST committed:
Change Climate set temp action for incorrect feature will raise (#126692)

* Change Climate set temp action for incorrect feature will raise

* Fix some tests

* Fix review comments

* Fix tesla_fleet

* Fix tests

* Fix review comment

zxdavb committed:
remove parameterization

starkillerOG committed:
Bump reolink-aio to 0.9.11 (#126778)

epenet committed:
Use shorthand attributes in geofency device tracker (#126741)

sdb9696 committed:
Deprecate tplink alarm button entities (#126349)

Co-authored-by: J. Nick Koston <nick@koston.org>

zxdavb committed:
Remove unnecessary patch from evohome tests (#126760)

starkillerOG committed:
Remove Reolink Home Hub main level switches (#126697)

Co-authored-by: Robert Resch <robert@resch.dev>

zxdavb committed:
clean snapshots

zxdavb commented:
Argh!
